ros2_easy_test/env.py

```diff
@@ -351,3 +393,66 @@ def call_service(
 
         future = client.call_async(request)
         return cast(SrvTypeResponse, self.await_future(future, timeout=timeout_call))
+
+    def _get_action_objects(self, name: str, action_type: Type) -> ActionObjects:
+        """Returns the service client for the given service name."""
+
+        with self._registered_actions_lock:
+            try:
+                return self._registered_actions[name]
+            except KeyError:
+                objects = ActionObjects(
+                    client=ActionClient(self, action_type, name, callback_group=self._action_cb_group)
+                )
+                self._registered_actions[name] = objects
+                return objects
+
+    def send_action_goal_and_wait_for_response(
+        self,
+        name: str,
+        action_type: Type,
+        goal_msg: Any,
+        timeout_availability: Optional[float] = 1,
+        timeout_send_goal: Optional[float] = 1,
+        timeout_get_result: Optional[float] = 10,
+    ) -> tuple[Optional[ClientGoalHandle], Optional[List[Any]], Optional[Any]]:
+        """Sends the goal to the given action and returns the response, feedbacks and result.
+
+        The action type is inferred automatically from the request type.
+
+        Args:
+            name: The name of the service
+            request: The request to send to the service
+            timeout_availability: The timeout to wait for the service to be available
+            timeout_call: The timeout to wait for the service to respond
+
+        Returns:
+            The response of the service
+
+        Raises:
+            TimeoutError: If the service did not respond within the timeout
+        """
+        action_objects = self._get_action_objects(name, action_type)
+
+        is_ready = action_objects.client.wait_for_server(timeout_availability)
+        if not is_ready:
+            raise TimeoutError(f"Action {name} did not become ready within {timeout_availability} seconds")
+
+        action_objects.send_goal_future = action_objects.client.send_goal_async(
+            goal_msg, feedback_callback=action_objects.feedback_callback
+        )
+        action_objects.send_goal_future.add_done_callback(action_objects.goal_response_callback)
+
+        action_objects.goal_handle = self.await_future(
+            action_objects.send_goal_future, timeout=timeout_send_goal
+        )
+
+        # Spin so we have a goal_response_callback and get_result_future is set.
+        while action_objects.get_result_future is None:
+            self.executor.spin_once()
+
+        action_objects.result = self.await_future(
+            action_objects.get_result_future, timeout=timeout_get_result
+        )
+
+        return action_objects.goal_handle, action_objects.feedbacks, action_objects.result
```

Comment:
As the first element, can't we directly return `action_objects.goal_handle.get_result()`? It should be set, right?

And the second element can never be None, right? Or if, we should return an empty list instead.

Reply:
If you see in test_actions, the handle returns both status and result, so we can verify both.
For feedbacks we initialize it with an empty list in the constructor so should always be a list.  There could be an action which doesn't provide feedback, so would return an empty list.

Reply:
> If you see in test_actions, the handle returns both status and result so that we can verify both.

Makes sense!

>  For feedbacks we initialize it with an empty list in the constructor so should always be a list. There could be an action which doesn't provide feedback, so would return an empty list.

Yes, exactly. So what I wanted to say was: We should change the return type of the function.